test/unit/provider/execute.js

```diff
@@ -0,0 +1,174 @@
+'use strict';
+
+const { join } = require('path');
+const metatests = require('metatests');
+const metaschema = require('metaschema');
+const {
+  errors: { ValidationError },
+} = metaschema;
+
+const { StorageProvider } = require('../../../lib/provider');
+const { options, config } = require('../../../lib/metaschema-config/config');
+const { GSError, codes: errorCodes } = require('../../../lib/errors');
+
+const schemasDir = join(__dirname, '../../fixtures/provider-execute');
+
+const CATEGORY = 'Category';
+const ACTION = 'Action';
+const PUBLIC_ACTION = 'PublicAction';
+const ACTION_ARGS = { a: 2, b: 1 };
+
+metatests.test('provider.execute unit test', async test => {
+  let schema;
+
+  try {
+    schema = await metaschema.fs.load(schemasDir, options, config);
+  } catch (err) {
+    test.fail(err);
+    test.end();
+    return;
+  }
+
+  const provider = new StorageProvider({});
+
+  provider.open({ schema }, () => {
```

Comment:
```suggestion
  provider.open({ schema }, test.cbFail(() => {
```
This will handle an error from `open` if it happens and `end` the test with it.

Reply:
@lundibundi I can do this only if I change `metatests`'s version from `^0.3.0` to `^0.5.0` in `package.json`  in order to use `cbFail`. Am I allowed to do this?